src/main/java/com/musala/atmosphere/commons/websocket/WebSocketCommunicatorManager.java

```diff
@@ -0,0 +1,60 @@
+package com.musala.atmosphere.commons.websocket;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.musala.atmosphere.commons.websocket.message.ResponseMessage;
+
+public class WebSocketCommunicatorManager {
+    private static WebSocketCommunicatorManager instance = null;
+
+    private static Map<String, Object> lockObjectMap;
+
+    private static Map<String, ResponseMessage> responseMap;
+
+    private WebSocketCommunicatorManager() {
+        lockObjectMap = new ConcurrentHashMap<String, Object>();
+        responseMap = new ConcurrentHashMap<String, ResponseMessage>();
+    }
+
+    public static WebSocketCommunicatorManager getInstance() {
+        if (instance == null) {
+            synchronized (WebSocketCommunicatorManager.class) {
+                if (instance == null) {
+                    instance = new WebSocketCommunicatorManager();
+                }
+            }
+        }
+
+        return instance;
+    }
+
+    public Object getSynchronizationObject(String sessionId) {
+        if (lockObjectMap.containsKey(sessionId)) {
+            return lockObjectMap.get(sessionId);
+        }
+
+        Object syncObj = new Object();
+        lockObjectMap.put(sessionId, syncObj);
+        return syncObj;
+    }
+
+    public ResponseMessage getServerResponse(String sessionId) {
```

Comment:
Everything in this method can be written as just 
```java
return responseMap.remove(sessionId);
```
I would still keep it in two rows, for the sake of readability:
```java
ResponseMessage response = responseMap.remove(sessionId);
return response;
```